test/integration/good/code-gen/cpp.expected

```diff
@@ -19567,54 +19010,20 @@ class param_constraint_model final : public model_base_crtp<param_constraint_mod
       L_Omega = std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>>(nt, Eigen::Matrix<local_scalar_t__, -1, -1>(2, 2));
       stan::math::fill(L_Omega, DUMMY_VAR__);
       
-      std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>> L_Omega_in__;
-      L_Omega_in__ = std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>>(nt, Eigen::Matrix<local_scalar_t__, -1, 1>(
-        ((2 * (2 - 1)) / 2)));
-      stan::math::fill(L_Omega_in__, DUMMY_VAR__);
-      
-      current_statement__ = 1;
-      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
-        current_statement__ = 1;
-        assign(L_Omega_in__, in__.vector(((2 * (2 - 1)) / 2)),
-          "assigning variable L_Omega_in__", index_uni(sym1__));}
       current_statement__ = 1;
-      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
-        current_statement__ = 1;
-        if (jacobian__) {
-          current_statement__ = 1;
-          assign(L_Omega,
-            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)],
-              2, lp__), "assigning variable L_Omega", index_uni(sym1__));
-        } else {
-          current_statement__ = 1;
-          assign(L_Omega,
-            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)], 2),
-            "assigning variable L_Omega", index_uni(sym1__));
-        }}
+      assign(L_Omega,
+        in__.template read_constrain_cholesky_factor_corr<std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>>, jacobian__>(
+          lp__, nt, 2, 2), "assigning variable L_Omega");
```

Comment:
Alright so fixing this bit here got it to `param-constrain.stan` to compile for me (after we add the little fix in this PR [here](https://github.com/stan-dev/stan/pull/3024) since we forgot to include the deserializer header lol)

The model is 

```stan
data {
  int nt;
  int NS;
}

parameters {
  cholesky_factor_corr[2] L_Omega[nt];
  vector<lower=L_Omega[1,1,2]>[NS] z1;
}
```

Is making the `nt, 2, 2)` where the deserializer signature the dimensions to just be `nt, 2)` ([here](https://github.com/stan-dev/stan/blob/develop/src/stan/io/deserializer.hpp#L790)) since then we grab a lower triangular of data with # of cells equal to `(K * (K - 1)) / 2)`. I think this is caused by get_dims in the expression gen stuff for `let read = ...`

https://github.com/stan-dev/stanc3/pull/856/files#diff-a1f5ff85d2b828c19a91e01d5c1e8caaf75c47e8af9bb205a3e0ed486a7e064fR233

Reply:
Oooo, it's possible that we could have a problem somewhere. `cholesky_factor_cov` supports both: `cholesky_factor_cov[M]` and `cholesky_factor_cov[M, N]` (according to [here](https://mc-stan.org/docs/2_26/reference-manual/vector-and-matrix-data-types.html)). I think it's the only matrix parameter like this (that has a non-square version), so watch out.

Reply:
And I say it's possible there could be a problem not cause I know -- just cause it is a little different and but sounds related to what you're talking about.

Reply:
We also do this pattern for `read_constrain/free_cholesky_factor_corr`, `read_constrain/free_cov_matrix`, and `read_constrain/free_corr_matrix`.

Reply:
Aight I just pushed something up that I think will gen the thing we want

https://github.com/stan-dev/stanc3/pull/856/commits/3eaab9eef69c9867ce963d4b857d5111d4e2605d

Reply:
Oh ooff Ben just seeing your comment. I think that's fine? The signature for `read_constrain_cholesky_factor_cov` is 

```cpp
read_constrain_cholesky_factor_cov(LP& lp, Eigen::Index M, Eigen::Index N) 
```

And in the thing I wrote we handle that the normal way, so hopefully should be fine